docs/getting-started.md

```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+```typescript
+  let tokenContract = ERC721.bind(event.address);
+  let artPieceTokenAddress = tokenContract.symbol();
 ```
 
+Note that there is an `ERC721Token` object that we are calling `bind` on. This is gathered from the subgraph manifest here:
+```
+  source:
+    address: "f87e31492faf9a91b02ee0deaad50d51d56d5d4d"
+    abi: ERC721
+```
 
-Along with `store.get()` you can use `.getString()`, `.getU256()`, etc... for getting property values from the entity.
+And, like `store` and `Entity`, this object data source can be imported from `graph-ts`. 
 
-## 4 Build
+## 4 Build and Deploy your Subgraph 
 
 ### 4.1 Build and deploy your mappings to IPFS
 In order to deploy your subgraph to your Graph Node, the subgraph manifest will first need to be built and deployed to IPFS (along with all linked files).
 
-Follow the instructions [here](https://ipfs.io/docs/getting-started/) to start a locally running IPFS daemon.
+To get an IPFS daemon working locally,do the following:
+* Download and install IPFS 
+* run `ipfs init`
+* run `ipfs daemon`
+
+If you get stuck, you can follow the instructions from the IPFS website [here](https://ipfs.io/docs/getting-started/).
+
+Once you've started your IPFS daemon you can run `yarn build-ipfs` in your subgraph directory (it assumes the default tcp port for the IPFS daemon). `yarn build-ipfs` should have been set up in section 3.3 when you went through the [graph-cli documentation](https://github.com/graphprotocol/graph-cli). This will compile your mappings, and deploy the mappings, schema and the subgraph manifest itself to IPFS. The result should be a single content hash, and this is your `subgraph ID`. It will also create a `/dist` folder. There is no need to upload this folder to your git repository, and you should not edit any of the files in there because they are auto generated. 
+
+To confirm on  ipfs, you can pass that content hash into `ipfs cat` to view your subgraph manifest with files paths replaced by IPLD links.
+
+### 4.2 Deploying Your Subgraph
+
+Deploying your subgraph means the subgraph is running, and sourcing data from Ethereum, transforming that data with your mappings, and storing it in the graph node. Note that a running subgraph can safely be stopped and started, and it will pick up where it has left off. 
+
+#### 4.2.1 Create your Postgres db 
+
+Make sure you have postgres installed. Then run the following commands:
+
+`initdb -D .postgres`
+`createdb <YOUR_POSTGRES_DB_NAME>`
+
+Name your db something relevant to your project, so you always know how to access it. 
+
+#### 4.2.3 Setting up your Ethereum Endpoint
+
+When you deploy your subgraph, you need to tell the subgraph where to start grabbing Ethereum data from. There are three typical ways you can grab data from Ethereum which the graph supports, which are Infura, local nodes, and ganache. 
+
+Note, the command you must run requires that you have `rust` and `cargo` installed.
```

Comment:
I wouldn't use "must" and I would explain what's the command for (e.g. The command for deploying your subgraph to The Graph assumes that you have `rust` and `cargo` installed... )